controller/apps_osioclient.go

```diff
@@ -62,54 +66,70 @@ func (osioclient *OSIOClientV1) GetNamespaceByType(ctx context.Context, userServ
 func (osioclient *OSIOClientV1) GetUserServices(ctx context.Context) (*app.UserService, error) {
 	resp, err := osioclient.wc.ShowUserService(ctx, witclient.ShowUserServicePath())
 	if err != nil {
-		return nil, err
+		log.Error(ctx, map[string]interface{}{
+			"ShowUserServicePath": witclient.ShowUserServicePath(),
+			"err": err,
+		}, "could not retrieve uses services from ", witclient.ShowUserServicePath())
+		return nil, errs.Wrapf(err, "could not retrieve uses services")
 	}
 
 	defer resp.Body.Close()
 
 	respBody, err := ioutil.ReadAll(resp.Body)
 
 	status := resp.StatusCode
-	if status == 404 {
+	if status == http.StatusNotFound {
+		// 404 Not Found is either a bad URL path or an invalid space.
 		return nil, nil
 	} else if status < 200 || status > 300 {
-		return nil, errors.New("Failed to GET " + witclient.ShowUserServicePath() + " due to status code " + string(status))
+		return nil, errs.Errorf("failed to GET %s due to status code %d", witclient.ShowUserServicePath(), status)
 	}
 
 	var respType app.UserServiceSingle
 	err = json.Unmarshal(respBody, &respType)
 	if err != nil {
-		return nil, err
+		log.Error(ctx, map[string]interface{}{
+			"json": respBody,
+			"err":  err,
+		}, "could not unmarshal UserServiceSingle JSON")
+		return nil, errs.Wrapf(err, "could not unmarshal UserServiceSingle JSON")
 	}
 	return respType.Data, nil
 }
 
 // GetSpaceByID - fetch space given UUID
 func (osioclient *OSIOClientV1) GetSpaceByID(ctx context.Context, spaceID uuid.UUID) (*app.Space, error) {
-	// there are two different uuid packages at play here:
-	//   github.com/satori/go.uuid and goadesign/goa/uuid.
-	// because of that, we fenerate our own URL to avoid issues for now.
-	urlpath := fmt.Sprintf("/api/spaces/%s", spaceID.String())
+
+	urlpath := witclient.ShowSpacePath(goauuid.UUID(spaceID))
+
 	resp, err := osioclient.wc.ShowSpace(ctx, urlpath, nil, nil)
 	if err != nil {
-		return nil, err
+		log.Error(ctx, map[string]interface{}{
+			"ShowSpacePath": witclient.ShowSpacePath(goauuid.UUID(spaceID)),
```

Comment:
we've standardized the naming of log variables as "lower_case"

Reply:
Changed. 
What about logging environment variables, for example: 

    log.Error(ctx, map[string]interface{}{
        "FABRIC8_WIT_API_URL": witURLStr,
        "err": err,
    }, "cannot parse FABRIC8_WIT_API_URL: %s", witURLStr)

Reply:
I think those are fine